README.md

```diff
@@ -22,26 +41,183 @@ allprojects {
 
-...Not yet ready. Under development. 
+Before you can use the Customer.io SDK, you need to initialize it. `CustomerIO` is a singleton: you'll create it once and re-use it across your application.
 
-# Contributing
+```kotlin
+class App : Application() {
+   override fun onCreate() {
+       super.onCreate()
+      val customerIO = CustomerIO.Builder(
+            siteId = "your-site-id",
+            apiKey = "your-api-key",
+            appContext = this
+        ).build()
+   }
+}
+```
+The Builder for CustomerIO exposes configuration options for features such `region`,
+`timeout`. 
+
+> A best practice is to initialize CustomerIO in the Application class, this way you will be able to access that instance from any part of your application using the instance() method:
+
+# Tracking
+
+### Dependency
+Tracking artifact is needed in order to use tracking features
+```groovy
+implementation 'io.customer.android:tracking:<version-here>'
+```
+## Identify a customer
+
+When you identify a customer, you:
+1. Add or update the customer's profile in your workspace.
+2. Save the customer's information on the device. Future calls to the SDK are linked to the last-identified customer. For example, after you identify a person, any events that you track are automatically associated with that person.
+
+You can only identify one customer at a time. The SDK "remembers" the most recently-identified customer.
+If you identify customer "A", and then call the identify function for customer "B", the SDK "forgets" customer "A" and assumes that customer "B" is the current app user. 
+
+[Learn more about the `Tracking` class][s-tracking].
+
+```kotlin
+CustomerIO.instance()
+            .identify(
+                identifier = "989388339",
+                attributes = mapOf("first_name" to "firstName")
+            ).enqueue { result ->
+                when (result) {
+                    is ErrorResult -> {
+                        // Handle error
+                    }
+                    is Success -> {
+                        // Handle success
+                    }
+                }
+            }
+```
+
+## Track a custom event
+
+Once you've identified a customer, you can use the `track` method to capture customer activity and send it Customer.io, optionally supplying event data with your request
+
+```kotlin
+CustomerIO.instance().track(
+                name = "logged_in",
+                attributes = mapOf("ip" to "127.0.0.1")
+            ).enqueue { result ->
+                when (result) {
+                    is ErrorResult -> {
+                        // Handle error
+                    }
+                    is Success -> {
+                        // Handle success
+                    }
+                }
+            }
+```
+
+`attributes` accepts the following data types:
+- Primitives (int, float, char...) and their boxed counterparts (Integer, Float, Character...).
+- Arrays, Collections, Lists, Sets, and Maps
+- Strings
+- Enums
```

Comment:
how are enums converted to JSON strings in the SDK?

Reply:
Oh, sorry. I meant what does the resulting JSON string look for an enum? If I was a customer reading this document, I would ask that question. 

If I was using this code:
```kotlin
CustomerIO.instance().trackEvent("edit profile", mapOf("section" to ProfileSection.PHOTO))
```

I would wonder, what does the resulting JSON string for attributes look like?

```json
{"section": ???}
```

Reply:
It would be exactly hor the enum is set unless the adapter is customized 

so in the mentioned case, it would look like this 

`{"section": "PHOTO"}`

Reply:
Ok. I suggest a comment in the docs mentioning that.

Reply:
Mind suggesting it again, because I feel like it got lost among other comments and updates to README.?